neuralmonkey/trainers/regularizers.py

```diff
@@ -0,0 +1,139 @@
+"""Variable regularizers.
+
+This module contains classes that can be used as a variable regularizers
+during training. All implementation should be derived from the Regularizer
+class.
+
+"""
+from typing import List
+
+import numpy as np
+import tensorflow as tf
+from typeguard import check_argument_types
+
+from neuralmonkey.logging import log
+
+
+class Regularizer:
+    """Base class for the regularizers."""
+
+    def __init__(self,
+                 name: str,
+                 weight: float) -> None:
+        """Create the regularizer.
+
+        Arguments:
+            name: Regularizer name.
+            weight: Weight of the regularization term.
+        """
+        check_argument_types()
+
+        self._name = name
+        self._weight = weight
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def weight(self) -> float:
+        return self._weight
+
+    def value(self, variables) -> float:
+        raise NotImplementedError("Abstract method")
+
+
+class L1Regularizer(Regularizer):
+    """L1 regularizer."""
+
+    def __init__(self,
+                 name: str = "train_l1",
+                 weight: float = 1.0e-8) -> None:
```

Comment:
už jsem to napsal níž - vyházet neopodstatněný defaultní hodnoty u name i u weight

pokud se nějaká hodnota doporučuje (což u L1 ani L2 neni ten případ), napsal bych to do docstringu.